lib/health/health.go

```diff
@@ -136,60 +173,30 @@ func (aggregator *HealthAggregator) Summary() *HealthReport {
 	aggregator.mutex.Lock()
 	defer aggregator.mutex.Unlock()
 
-	var failedLivenessChecks []*reporterState
-	var failedReadinessChecks []*reporterState
-
 	// In the absence of any reporters, default to indicating that we are both live and ready.
 	summary := &HealthReport{Live: true, Ready: true}
 
 	// Now for each reporter...
 	for _, reporter := range aggregator.reporters {
-		// Reset Live to false if that reporter is registered to report liveness and hasn't
-		// recently said that it is live.
-		stillLive := reporter.latest.Live && !reporter.TimedOut()
-		if summary.Live && reporter.reports.Live && !stillLive {
+		log.WithField("reporter", reporter).Debug("Checking state of reporter")
+		if !reporter.RecentlyLive() {
```

Comment:
With this polarity and naming, it feels confusing to me that `reporter.RecentlyLive()` will be `true` if the reporter doesn't report liveness.  I think I would prefer one of these options:

1. Keep the `reporter.reports.Live` condition here, outside of `reporter.RecentlyLive()`.

2. Invert the polarity and naming, to (e.g.) `reporter.HasLivenessProblem()`.

And obviously similarly for readiness.

Reply:
(2) sounds good to me, reads very nicely